### DISCUSSION TITLE
time delay and toggle

### DISCUSSION THREAD

**Discussion by KD-4-Lyfe** [julianschill/klipper-led_effect#221] [Q&A]
is there a way to have an on/off toggle preferably the the on toggle has a time delay of 30seconds to turn off.

otherwise can i setup a time delay somehow besides using a G4 P### gcode

**Comment by julianschill** [comment]
I think the "delayed_gcode" function of Klipper might solve your issue.

**Reply by KD-4-Lyfe** [reply]
I found out chatGPT does some amazing work!
This is what I came up with if anyone else is looking.

[gcode_macro TOGGLE_WHITE_LIGHT]
variable_led_on: 0      # Initialize variable to track if the LED is on or off

gcode:
    {% if printer["gcode_macro TOGGLE_WHITE_LIGHT"].led_on == 0 %}
        SET_LED_EFFECT EFFECT=white_light    # Turn on white light
        SET_GCODE_VARIABLE MACRO=TOGGLE_WHITE_LIGHT VARIABLE=led_on VALUE=1
        UPDATE_DELAYED_GCODE ID=WHITE_LIGHT_off DURATION=10
    {% else %}
        SET_LED_EFFECT EFFECT=white_light stop=1    # Turn on white light
        SET_GCODE_VARIABLE MACRO=TOGGLE_WHITE_LIGHT VARIABLE=led_on VALUE=0
    {% endif %}

[delayed_gcode WHITE_LIGHT_off]
gcode:
    SET_LED_EFFECT EFFECT=white_light stop=1    # Turn off white light
    SET_GCODE_VARIABLE MACRO=TOGGLE_WHITE_LIGHT VARIABLE=led_on VALUE=0
